File: src/app/Class/user/UserProfile.ts
export class UserProfile {

    private userName: string;
    private email: string;
    private firstName: string;
    private lastName: string;
    private birthDate: Date;
    private aboutUser: string;


  constructor(obj?: any) {
      this.userName = (obj && obj.userName) || '';
      this.email = (obj && obj.email) || '';
      this.firstName = (obj && obj.firstName) || '';
      this.lastName = (obj && obj.lastName) || '';
      this.birthDate = (obj && obj.birthDate) || null;
      this.aboutUser = (obj && obj.aboutUser) || '';
  }

  // Getters

  getUserName(): string {
    return this.userName;
  }

  getEmail(): string {
    return this.email;
  }

  getFirstName(): string {
    return this.firstName;
  }

  getLastName(): string {
    return this.lastName;
  }

  getBirthDate(): Date {
    return this.birthDate;
  }

  getAboutUser(): string {
    return this.aboutUser;
  }

  // Setters

  setUserName(value: string) {
    this.userName = value;
  }

  setEmail(value: string) {
    this.email = value;
  }
  setFirstName(value: string) {
    this.firstName = value;
  }

  setLastName(value: string) {
    this.lastName = value;
  }

  setBirthDate(value: Date) {
    this.birthDate = value;
  }

  setAboutUser(value: string) {
    this.aboutUser = value;
  }

}

File: src/main/java/MusicCube/user/UserProfile.java
package MusicCube.user;

import MusicCube.entities.User;

import java.util.Date;

public class UserProfile {

    private int id;
    private String userName;
    private String email;
    private String firstName;
    private String lastName;
    private Date birthDate;
    private String aboutUser;

    // Constructors

    public UserProfile() {
    }

    public UserProfile(User user) {
        this.id = user.getId();
        this.userName = user.getUserName();
        this.email = user.getEmail();
        this.firstName = user.getFirstName();
        this.lastName = user.getLastName();
        this.birthDate = user.getBirthDate();
        this.aboutUser = user.getAboutUser();
    }

    public UserProfile(int id, String userName, String email, String firstName, String lastName, Date birthDate, String aboutUser) {
        this.id = id;
        this.userName = userName;
        this.email = email;
        this.firstName = firstName;
        this.lastName = lastName;
        this.birthDate = birthDate;
        this.aboutUser = aboutUser;
    }

// Getters and Setters

    public int getId() {
        return id;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Date getBirthDate() {
        return birthDate;
    }

    public void setBirthDate(Date birthDate) {
        this.birthDate = birthDate;
    }

    public String getAboutUser() {
        return aboutUser;
    }

    public void setAboutUser(String aboutUser) {
        this.aboutUser = aboutUser;
    }
}
